
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="A Snakemake-based workflow for quality control and detection of SV in Strand-Seq sequencing data">
      
      
      
      
        <link rel="prev" href="../usage/">
      
      
        <link rel="next" href="../publication/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.5.3, mkdocs-material-9.5.17">
    
    
      
        <title>Parameters - MosaiCatcher documentation</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.bcfcd587.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="black" data-md-color-accent="black">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#parameters" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="MosaiCatcher documentation" class="md-header__button md-logo" aria-label="MosaiCatcher documentation" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            MosaiCatcher documentation
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Parameters
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="black" data-md-color-accent="black"  aria-hidden="true"  type="radio" name="__palette" id="__palette_0">
    
  
</form>
      
    
    
      <script>var media,input,key,value,palette=__md_get("__palette");if(palette&&palette.color){"(prefers-color-scheme)"===palette.color.media&&(media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']"),palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent"));for([key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="MosaiCatcher documentation" class="md-nav__button md-logo" aria-label="MosaiCatcher documentation" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    MosaiCatcher documentation
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../installation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Installation
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../usage/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Usage
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Parameters
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Parameters
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#mosaicatcher-arguments" class="md-nav__link">
    <span class="md-ellipsis">
      MosaiCatcher arguments
    </span>
  </a>
  
    <nav class="md-nav" aria-label="MosaiCatcher arguments">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#general-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      General parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#data-location-inputoutput-options" class="md-nav__link">
    <span class="md-ellipsis">
      Data location &amp; Input/output options
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ashleys-qc-upstream-pipeline" class="md-nav__link">
    <span class="md-ellipsis">
      Ashleys-QC upstream pipeline
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#reference-data-chromosomes" class="md-nav__link">
    <span class="md-ellipsis">
      Reference data &amp; Chromosomes
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#counts-configuration" class="md-nav__link">
    <span class="md-ellipsis">
      Counts configuration
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sv-calling-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      SV calling parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sv-calling-algorithm-processing-options" class="md-nav__link">
    <span class="md-ellipsis">
      SV calling algorithm processing options
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#downstream-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      Downstream analysis
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#embl-specific-options" class="md-nav__link">
    <span class="md-ellipsis">
      EMBL specific options
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#execution-profile" class="md-nav__link">
    <span class="md-ellipsis">
      Execution profile
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#snakemake-arguments" class="md-nav__link">
    <span class="md-ellipsis">
      Snakemake arguments
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../publication/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Reproduce publication results
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../outputs/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Outputs
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../contributing/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Contributing
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#mosaicatcher-arguments" class="md-nav__link">
    <span class="md-ellipsis">
      MosaiCatcher arguments
    </span>
  </a>
  
    <nav class="md-nav" aria-label="MosaiCatcher arguments">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#general-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      General parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#data-location-inputoutput-options" class="md-nav__link">
    <span class="md-ellipsis">
      Data location &amp; Input/output options
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ashleys-qc-upstream-pipeline" class="md-nav__link">
    <span class="md-ellipsis">
      Ashleys-QC upstream pipeline
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#reference-data-chromosomes" class="md-nav__link">
    <span class="md-ellipsis">
      Reference data &amp; Chromosomes
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#counts-configuration" class="md-nav__link">
    <span class="md-ellipsis">
      Counts configuration
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sv-calling-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      SV calling parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sv-calling-algorithm-processing-options" class="md-nav__link">
    <span class="md-ellipsis">
      SV calling algorithm processing options
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#downstream-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      Downstream analysis
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#embl-specific-options" class="md-nav__link">
    <span class="md-ellipsis">
      EMBL specific options
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#execution-profile" class="md-nav__link">
    <span class="md-ellipsis">
      Execution profile
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#snakemake-arguments" class="md-nav__link">
    <span class="md-ellipsis">
      Snakemake arguments
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h1 id="parameters">Parameters</h1>
<h2 id="mosaicatcher-arguments">MosaiCatcher arguments</h2>
<div class="admonition note">
<p class="admonition-title">How to pass configuration arguments?</p>
<p>All these arguments can be specified in two ways:</p>
<ol>
<li>In the config/config.yaml file, by replacing existing values</li>
<li>Using the <code>--config</code> snakemake argument (<code>--config</code> must be called only one time with all the arguments behind it, e.g: <code>--config input_bam_location=&lt;INPUT&gt; output_location=&lt;OUTPUT&gt; email=&lt;EMAIL&gt;</code>)</li>
</ol>
</div>
<h3 id="general-parameters">General parameters</h3>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Comment</th>
<th>Default</th>
<th>Example</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>email</code></td>
<td>Email address for completion summary</td>
<td>None</td>
<td>None</td>
</tr>
</tbody>
</table>
<h3 id="data-location-inputoutput-options">Data location &amp; Input/output options</h3>
<style>
table th:first-of-type {
    width: 30%;
}
table th:nth-of-type(2) {
    width: 30%;
}
table th:nth-of-type(3) {
    width: 10%;
}
table th:nth-of-type(4) {
    width: 20%;
}
</style>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Comment</th>
<th>Parameter type</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>data_location</code></td>
<td>Path to parent folder containing samples</td>
<td>String</td>
<td>.tests/data_CHR17/</td>
</tr>
<tr>
<td><code>samples_to_process</code></td>
<td>If multiple plates in the data_location parent folder, specify one or a comma-sep list of samples</td>
<td>None</td>
<td>"[SampleA,SampleB]"</td>
</tr>
<tr>
<td><code>publishdir</code></td>
<td>Path to backup location where important data is copied</td>
<td>String</td>
<td></td>
</tr>
</tbody>
</table>
<h3 id="ashleys-qc-upstream-pipeline">Ashleys-QC upstream pipeline</h3>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Comment</th>
<th>Parameter type</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>input_bam_legacy</code></td>
<td>Mutualy exclusive with ashleys_pipeline. Will use <code>selected</code> folder to identify high-quality libraries to process</td>
<td>Boolean</td>
<td>False</td>
</tr>
<tr>
<td><code>ashleys_pipeline</code></td>
<td>Allow to load and use <a href="/docs/usage.md#3c-fastq-input--preprocessing-module">ashleys-qc-pipeline</a> snakemake preprocessing module and to start from FASTQ inputs</td>
<td>Boolean</td>
<td>False</td>
</tr>
<tr>
<td><code>ashleys_pipeline_only</code></td>
<td>Stop the execution after ashleys-qc-pipeline submodule. Requires <code>ashleys_pipeline</code> to be True</td>
<td>Boolean</td>
<td>False</td>
</tr>
<tr>
<td><code>ashleys_threshold</code></td>
<td>Threshold for Ashleys-qc binary classification</td>
<td>Float</td>
<td>0.5</td>
</tr>
<tr>
<td><code>bypass_ashleys</code></td>
<td>Set all cells as high-quality (labels to 1)</td>
<td>False</td>
<td></td>
</tr>
<tr>
<td><code>MultiQC</code></td>
<td>Enable or disable MultiQC analysis (includes FastQC, samtools flagstats &amp; idxstats)</td>
<td>Boolean</td>
<td>False</td>
</tr>
<tr>
<td><code>hand_selection</code></td>
<td>Enable or disable hand selection through the Jupyter Notebook</td>
<td>Boolean</td>
<td>False</td>
</tr>
<tr>
<td><code>split_qc_plot</code></td>
<td>Enable or disable the split of QC plot into individual pages plots</td>
<td>Boolean</td>
<td>False</td>
</tr>
</tbody>
</table>
<h3 id="reference-data-chromosomes">Reference data &amp; Chromosomes</h3>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Comment</th>
<th>Default (options)</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>reference</code></td>
<td>Reference genome</td>
<td>hg38 (hg19, T2T, mm10)</td>
</tr>
<tr>
<td><code>chromosomes</code></td>
<td>List of chromosomes to be processed</td>
<td>Human: chr[1..22,X,Y], Mouse: chr[1..20,X,Y]</td>
</tr>
<tr>
<td><code>chromosomes_to_exclude</code></td>
<td>List of chromosomes to exclude</td>
<td>[]</td>
</tr>
</tbody>
</table>
<h3 id="counts-configuration">Counts configuration</h3>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Comment</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>multistep_normalisation</code></td>
<td>Allow to perform <a href="/docs/usage.md#multistep-normalisation">multistep normalisation</a> including GC correction for visualization (Marco Cosenza).</td>
<td>False</td>
</tr>
<tr>
<td><code>window</code></td>
<td>Window size used for binning by mosaic count (Can be of high importance regarding library coverage)</td>
<td>100000</td>
</tr>
<tr>
<td><code>blacklist_regions</code></td>
<td>Enable/Disable blacklisting</td>
<td>True</td>
</tr>
</tbody>
</table>
<h3 id="sv-calling-parameters">SV calling parameters</h3>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Comment</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>multistep_normalisation_for_SV_calling</code></td>
<td>Allow to use multistep normalisation count file during SV calling (Marco Cosenza).</td>
<td>False</td>
</tr>
<tr>
<td><code>hgsvc_based_normalized_counts</code></td>
<td>Use HGSVC based normalisation .</td>
<td>True</td>
</tr>
</tbody>
</table>
<h3 id="sv-calling-algorithm-processing-options">SV calling algorithm processing options</h3>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Comment</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>min_diff_jointseg</code></td>
<td>Minimum difference in error term to include another breakpoint in the joint segmentation (default=0.5)</td>
<td>0.1</td>
</tr>
<tr>
<td><code>min_diff_singleseg</code></td>
<td>Minimum difference in error term to include another breakpoint in the single-cell segmentation (default=1)</td>
<td>0.5</td>
</tr>
<tr>
<td><code>additional_sce_cutoff</code></td>
<td>Minimum gain in mismatch distance needed to add an additional SCE</td>
<td>20000000</td>
</tr>
<tr>
<td><code>sce_min_distance</code></td>
<td>Minimum distance of an SCE to a break in the joint segmentation</td>
<td>500000</td>
</tr>
<tr>
<td><code>llr</code></td>
<td>Likelihood ratio used to detect SV calls</td>
<td>4</td>
</tr>
</tbody>
</table>
<h3 id="downstream-analysis">Downstream analysis</h3>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Comment</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>arbigent</code></td>
<td>Enable <a href="/docs/usage.md#arbigent-mode-of-execution">ArbiGent</a> mode of execution to genotype SV based on arbitrary segments</td>
<td>False</td>
</tr>
<tr>
<td><code>arbigent_bed_file</code></td>
<td>Allow to specify custom ArbiGent BED file</td>
<td>""</td>
</tr>
<tr>
<td><code>scNOVA</code></td>
<td>Enable <a href="/docs/usage.md#scnova-mode-of-execution">scNOVA</a> mode of execution to compute Nucleosome Occupancy (NO) of detected SV</td>
<td>False</td>
</tr>
<tr>
<td><code>scTRIP_multiplot</code></td>
<td>Enable <a href="/docs/usage.md#sctrip-multiplot">scTRIP multiplot</a> generation for all chromosomes of all cells</td>
<td>False</td>
</tr>
</tbody>
</table>
<h3 id="embl-specific-options">EMBL specific options</h3>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Comment</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>genecore</code></td>
<td>Enable/disable genecore mode to give as input the genecore shared folder in /g/korbel/shared/genecore</td>
<td>False</td>
</tr>
<tr>
<td><code>genecore_date_folder</code></td>
<td>Specify folder to be processed</td>
<td></td>
</tr>
<tr>
<td><code>genecore_prefix</code></td>
<td>Specify genecore prefix folder</td>
<td>/g/korbel/STOCKS/Data/Assay/sequencing/2023</td>
</tr>
<tr>
<td><code>genecore_regex_elements</code></td>
<td>Specify genecore regex element to be used to distinguish sample from well number</td>
<td>PE20</td>
</tr>
</tbody>
</table>
<p>If <code>genecore</code> and <code>genecore_date_folder</code> are correctly specified, each plate will be processed independently by creating a specific folder in the <code>data_location</code> folder.</p>
<h3 id="execution-profile">Execution profile</h3>
<p><em>Location</em>: workflow/snakemake_profiles/</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Comment</th>
<th>Conda</th>
<th>Singularity</th>
<th>HPC</th>
<th>Local</th>
</tr>
</thead>
<tbody>
<tr>
<td>local/conda</td>
<td>/</td>
<td>X</td>
<td></td>
<td></td>
<td>X</td>
</tr>
<tr>
<td>local/conda_singularity</td>
<td>/</td>
<td>X</td>
<td>X</td>
<td></td>
<td>X</td>
</tr>
<tr>
<td>HPC/slurm_generic (to modify)</td>
<td>/</td>
<td>X</td>
<td></td>
<td>X</td>
<td></td>
</tr>
<tr>
<td>HPC/slurm_EMBL (optimised for EMBL HPC)</td>
<td>/</td>
<td>X</td>
<td></td>
<td>X</td>
<td></td>
</tr>
</tbody>
</table>
<h2 id="snakemake-arguments">Snakemake arguments</h2>
<p>Here are presented some essential snakemake options that could help you.</p>
<div class="highlight"><pre><span></span><code>--cores, -c
</code></pre></div>
<p>Use at most N CPU cores/jobs in parallel. If N is omitted or ‘all’, the limit is set to the number of available CPU cores. In case of cluster/cloud execution, this argument sets the number of total cores used over all jobs (made available to rules via workflow.cores).</p>
<div class="highlight"><pre><span></span><code>--printshellcmds, -p
</code></pre></div>
<p>Recommended to print out the shell commands that will be executed.</p>
<div class="highlight"><pre><span></span><code>--use-conda
</code></pre></div>
<p>If defined in the rule, run job in a conda environment. If this flag is not set, the conda directive is ignored and use the current environment (and path system) to execute the command.</p>
<div class="highlight"><pre><span></span><code>--conda-frontend [mamba|conda]
</code></pre></div>
<p>Choose the conda frontend for installing environments. Mamba is much faster and highly recommended but could not be installed by default on your system. Default: “conda”</p>
<div class="highlight"><pre><span></span><code>--use-singularity
</code></pre></div>
<p>If defined in the rule, run job within a singularity container. If this flag is not set, the singularity directive is ignored.</p>
<div class="highlight"><pre><span></span><code>--singularity-args &quot;-B /mounting_point:/mounting_point&quot;
</code></pre></div>
<p>Pass additional args to singularity. <code>-B</code> stands for binding point between the host and the container.</p>
<div class="highlight"><pre><span></span><code>--dryrun, -n
</code></pre></div>
<p>Do not execute anything, and display what would be done. If you have a very large workflow, use –dry-run –quiet to just print a summary of the DAG of jobs.</p>
<div class="highlight"><pre><span></span><code>--rerun-incomplete, --ri
</code></pre></div>
<p>Re-run all jobs the output of which is recognized as incomplete.</p>
<div class="highlight"><pre><span></span><code>--keep-going, -k
</code></pre></div>
<p>Go on with independent jobs if a job fails.</p>
<div class="highlight"><pre><span></span><code>-T, --retries, --restart-times
</code></pre></div>
<p>Number of times to restart failing jobs (defaults to 0).</p>
<div class="highlight"><pre><span></span><code>--forceall, -F
</code></pre></div>
<p>Force the execution of the selected (or the first) rule and all rules it is dependent on regardless of already created output.</p>
<hr />
<p><strong>ℹ️ Note</strong></p>
<p>Currently, the binding command needs to correspond to the mounting point of your system (i.e: "/tmp:/tmp").
On seneca for example (EMBL), use <code>"/g:/g"</code> if you are working on <code>/g/korbel[2]</code> or <code>"/scratch:/scratch"</code> if you plan to work on <code>scratch</code>.</p>
<hr />
<p>Obviously, all other <a href="https://snakemake.readthedocs.io/en/stable/executing/cli.html">snakemake CLI options</a> can also be used.</p>
<hr />












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "..", "features": ["navigation.sections"], "search": "../assets/javascripts/workers/search.b8dbb3d2.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.1e8ae164.min.js"></script>
      
    
  </body>
</html>