
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="A Snakemake-based workflow for quality control and detection of SV in Strand-Seq sequencing data">
      
      
      
      
      
        <link rel="next" href="installation/">
      
      
      <link rel="icon" href="assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.5.3, mkdocs-material-9.5.17">
    
    
      
        <title>MosaiCatcher documentation</title>
      
    
    
      <link rel="stylesheet" href="assets/stylesheets/main.bcfcd587.min.css">
      
        
        <link rel="stylesheet" href="assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL(".",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="white">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#overview" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
      </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="." title="MosaiCatcher documentation" class="md-header__button md-logo" aria-label="MosaiCatcher documentation" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            MosaiCatcher documentation
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Home
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="white"  aria-hidden="true"  type="radio" name="__palette" id="__palette_0">
    
  
</form>
      
    
    
      <script>var media,input,key,value,palette=__md_get("__palette");if(palette&&palette.color){"(prefers-color-scheme)"===palette.color.media&&(media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']"),palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent"));for([key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="." title="MosaiCatcher documentation" class="md-nav__button md-logo" aria-label="MosaiCatcher documentation" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    MosaiCatcher documentation
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Home
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="." class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#ashleys-qc-pipeline" class="md-nav__link">
    <span class="md-ellipsis">
      Ashleys-QC Pipeline
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mosaicatcher" class="md-nav__link">
    <span class="md-ellipsis">
      MosaiCatcher
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#roadmap" class="md-nav__link">
    <span class="md-ellipsis">
      📆 Roadmap
    </span>
  </a>
  
    <nav class="md-nav" aria-label="📆 Roadmap">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#technical-related-features" class="md-nav__link">
    <span class="md-ellipsis">
      Technical-related features
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bioinformatic-related-features" class="md-nav__link">
    <span class="md-ellipsis">
      Bioinformatic-related features
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#small-issues-to-fix" class="md-nav__link">
    <span class="md-ellipsis">
      Small issues to fix
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#troubleshooting-current-limitations" class="md-nav__link">
    <span class="md-ellipsis">
      🛑 Troubleshooting &amp; Current limitations
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#authors-alphabetical-order" class="md-nav__link">
    <span class="md-ellipsis">
      💂‍♂️ Authors (alphabetical order)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#citing-mosaicatcher" class="md-nav__link">
    <span class="md-ellipsis">
      Citing MosaiCatcher
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    <span class="md-ellipsis">
      📕 References
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="installation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Installation
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="usage/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Usage
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="parameters/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Parameters
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="publication/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Reproduce publication results
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="outputs/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Outputs
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="contributing/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Contributing
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#ashleys-qc-pipeline" class="md-nav__link">
    <span class="md-ellipsis">
      Ashleys-QC Pipeline
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mosaicatcher" class="md-nav__link">
    <span class="md-ellipsis">
      MosaiCatcher
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#roadmap" class="md-nav__link">
    <span class="md-ellipsis">
      📆 Roadmap
    </span>
  </a>
  
    <nav class="md-nav" aria-label="📆 Roadmap">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#technical-related-features" class="md-nav__link">
    <span class="md-ellipsis">
      Technical-related features
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bioinformatic-related-features" class="md-nav__link">
    <span class="md-ellipsis">
      Bioinformatic-related features
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#small-issues-to-fix" class="md-nav__link">
    <span class="md-ellipsis">
      Small issues to fix
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#troubleshooting-current-limitations" class="md-nav__link">
    <span class="md-ellipsis">
      🛑 Troubleshooting &amp; Current limitations
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#authors-alphabetical-order" class="md-nav__link">
    <span class="md-ellipsis">
      💂‍♂️ Authors (alphabetical order)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#citing-mosaicatcher" class="md-nav__link">
    <span class="md-ellipsis">
      Citing MosaiCatcher
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    <span class="md-ellipsis">
      📕 References
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h1 id="overview">Overview</h1>
<p><img alt="summary" src="images/figure_pipeline.png" /></p>
<div class="admonition note">
<p class="admonition-title">ashleys-qc-pipeline &amp; mosaicatcher-pipeline dependencies</p>
<p>From 2.2.0, you don't need to clone both <a href="https://github.com/friendsofstrandseq/ashleys-qc-pipeline">ashleys-qc-pipeline preprocessing module</a> and <a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline">mosaicatcher-pipeline</a>. By using <code>ashleys_pipeline_only=True</code> combined with <code>ashleys_pipeline=True</code> in the configuration of MosaiCatcher, this will stop the execution after the generation of the files related to ashleys-qc-pipeline. This allow you to use a single pipeline, repository and container and limit the potential conflicts by processing the same folder (data_location) by different repositories and set of files (including the workflow/data/ref_genomes references files).</p>
</div>
<h2 id="ashleys-qc-pipeline">Ashleys-QC Pipeline</h2>
<p>This workflow uses <a href="https://github.com/snakemake/snakemake">Snakemake</a> perform Quality Control analysis on Strand-Seq single-cell
sequencing data. The starting point are single-cell FASTQ files from Strand-seq experiments and the final output produced is a folder with clean selected BAM files. The pipeline can identify automatically high-quality libraries through ML-based analysis tool <a href="https://github.com/friendsofstrandseq/ashleys-qc">ashleys-qc</a>. Thus, the workflow goes through the following steps:</p>
<ol>
<li>FASTQ sequencing Quality Control through <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a></li>
<li>Mapping FASTQ against a reference genome throught <a href="http://bio-bwa.sourceforge.net/">BWA</a></li>
<li>Sorting, Deduplicating and Indexing of BAM files through <a href="http://www.htslib.org/">Samtools</a> &amp; <a href="https://lomereiter.github.io/sambamba/docs/sambamba-view.html">sambaba</a></li>
<li>Generating features and use <a href="https://github.com/friendsofstrandseq/ashleys-qc">ashleys-qc</a> model to identify high-quality cells</li>
</ol>
<h2 id="mosaicatcher">MosaiCatcher</h2>
<p>The starting point in MosaiCatcher are single-cell
BAM files from Strand-seq experiments and the final output are SV predictions in
a tabular format as well as in a graphical representation. To get to this point,
the workflow goes through the following steps:</p>
<ol>
<li>Binning of sequencing reads in genomic windows of 200kb via <a href="https://github.com/friendsofstrandseq/mosaicatcher">mosaic</a></li>
<li>Strand state detection</li>
<li>[Optional] Normalization of coverage with respect to a reference sample</li>
<li>Multi-variate segmentation of cells (<a href="https://github.com/friendsofstrandseq/mosaicatcher">mosaic</a>)</li>
<li>Haplotype resolution via <a href="https://github.com/daewoooo/StrandPhaseR">StrandPhaseR</a></li>
<li>Bayesian classification of segmentation to find SVs using MosaiClassifier</li>
<li>Visualization of results using custom R plots</li>
</ol>
<h2 id="roadmap">📆 Roadmap</h2>
<h3 id="technical-related-features">Technical-related features</h3>
<ul class="task-list">
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> Zenodo automatic download of external files + indexes (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/1.2.1">1.2.1</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> Multiple samples in the parent folder (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/1.2.2">1.2.2</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> Automatic testing of BAM SM tag compared to sample folder name (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/1.2.3">1.2.3</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> On-error/success e-mail (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/1.3">1.3</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> HPC execution (slurm profile for the moment) (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/1.3">1.3</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> Full singularity image with preinstalled conda envs (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/1.5.1">1.5.1</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> Single BAM folder with side config file (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/1.6.1">1.6.1</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> (EMBL) GeneCore mode of execution: allow selection and execution directly by specifying genecore run folder (2022-11-02-H372MAFX5 for instance) (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/1.8.2">1.8.2</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> Version synchronisation between ashleys-qc-pipeline and mosaicatcher-pipeline (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/1.8.3">1.8.3</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> Report captions update (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/1.8.5">1.8.5</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> Clustering plot (heatmap) &amp; SV calls plot update (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/1.8.6">1.8.6</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> <a href="/docs/usage.md#usage"><code>ashleys_pipeline_only</code> parameter</a>: using mosaicatcher-pipeline, trigger ashleys-qc-pipeline only and will stop after the generation of the counts, ashleys predictions &amp; plots to allow the user manual reviewing/selection of the cells to be processed (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/2.2.0">2.2.0</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled/><span class="task-list-indicator"></span></label> Plotting options (enable/disable segmentation back colors)</li>
</ul>
<h3 id="bioinformatic-related-features">Bioinformatic-related features</h3>
<ul class="task-list">
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> Self-handling of low-coverage cells (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/1.6.1">1.6.1</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> Upstream <a href="https://github.com/friendsofstrandseq/ashleys-qc-pipeline.git">ashleys-qc-pipeline</a> and FASTQ handle (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/1.6.1">1.6.1</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> Change of reference genome (currently only GRCh38) (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/1.7.0">1.7.0</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> Ploidy detection at the segment and the chromosome level: used to bypass StrandPhaseR if more than half of a chromosome is haploid (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/1.7.0">1.7.0</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> inpub_bam_legacy mode (bam/selected folders) (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/1.8.4">1.8.4</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> Blacklist regions files for T2T &amp; hg19 (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/1.8.5">1.8.5</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> <a href="/docs/usage.md#arbigent-mode-of-execution">ArbiGent</a> integration: Strand-Seq based genotyper to study SV containly at least 500bp of uniquely mappable sequence (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/1.9.0">1.9.0</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> <a href="/docs/usage.md#scnova-mode-of-execution">scNOVA</a> integration: Strand-Seq Single-Cell Nucleosome Occupancy and genetic Variation Analysis (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/1.9.2">1.9.2</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> <a href="/docs/usage.md#multistep-normalisation"><code>multistep_normalisation</code> and <code>multistep_normalisation_for_SV_calling</code> parameters</a> to replace GC analysis module (library size normalisation, GC correction, Variance Stabilising Transformation) (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/2.1.1">2.1.1</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> Strand-Seq processing based on mm10 assembly (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/2.1.2">2.1.2</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> UCSC ready to use file generation including counts &amp; SV calls (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/2.1.2">2.1.2</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> <code>blacklist_regions</code> parameter: (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/2.2.0">2.2.0</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> IGV ready to use XML session generation: (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/2.2.2">2.2.2</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled/><span class="task-list-indicator"></span></label> Pooled samples</li>
</ul>
<h3 id="small-issues-to-fix">Small issues to fix</h3>
<ul class="task-list">
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> replace <code>input_bam_location</code> by <code>data_location</code> (harmonization with <a href="https://github.com/friendsofstrandseq/ashleys-qc-pipeline.git">ashleys-qc-pipeline</a>)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> List of commands available through list_commands parameter (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/1.8.6">1.8.6</a></li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> Move pysam / SM tag comparison script to snakemake rule (<a href="https://github.com/friendsofstrandseq/mosaicatcher-pipeline/releases/tag/2.2.0">2.2.0</a>)</li>
</ul>
<h2 id="troubleshooting-current-limitations">🛑 Troubleshooting &amp; Current limitations</h2>
<ul>
<li>Do not change the structure of your input folder after running the pipeline, first execution will build a config dataframe file (<code>OUTPUT_DIRECTORY/config/config.tsv</code>) that contains the list of cells and the associated paths</li>
<li>Do not change the list of chromosomes after a first execution (i.e: first execution on <code>chr17</code>, second execution on all chromosomes)</li>
</ul>
<h2 id="authors-alphabetical-order">💂‍♂️ Authors (alphabetical order)</h2>
<ul>
<li>Ashraf Hufash</li>
<li>Cosenza Marco</li>
<li>Ebert Peter</li>
<li>Ghareghani Maryam</li>
<li>Grimes Karen</li>
<li>Gros Christina</li>
<li>Höps Wolfram</li>
<li>Jeong Hyobin</li>
<li>Kinanen Venla</li>
<li>Korbel Jan</li>
<li>Marschall Tobias</li>
<li>Meiers Sasha</li>
<li>Porubsky David</li>
<li>Rausch Tobias</li>
<li>Sanders Ashley</li>
<li>Van Vliet Alex</li>
<li>Weber Thomas (maintainer and current developer)</li>
</ul>
<h2 id="citing-mosaicatcher">Citing MosaiCatcher</h2>
<p>When using MosaiCatcher for a publication, please <strong>cite the following article</strong> in your paper:</p>
<p><a href="https://doi.org/10.1093/bioinformatics/btad633">MosaiCatcher v2 publication: Weber Thomas, Marco Raffaele Cosenza, and Jan Korbel. 2023. ‘MosaiCatcher v2: A Single-Cell Structural Variations Detection and Analysis Reference Framework Based on Strand-Seq’. Bioinformatics 39 (11): btad633. https://doi.org/10.1093/bioinformatics/btad633.</a></p>
<h2 id="references">📕 References</h2>
<blockquote>
<p>MosaiCatcher v2 publication: Weber Thomas, Marco Raffaele Cosenza, and Jan Korbel. 2023. ‘MosaiCatcher v2: A Single-Cell Structural Variations Detection and Analysis Reference Framework Based on Strand-Seq’. Bioinformatics 39 (11): btad633. <a href="https://doi.org/10.1093/bioinformatics/btad633">https://doi.org/10.1093/bioinformatics/btad633</a></p>
<p>Strand-seq publication: Falconer, E., Hills, M., Naumann, U. et al. DNA template strand sequencing of single-cells maps genomic rearrangements at high resolution. Nat Methods 9, 1107–1112 (2012). <a href="https://doi.org/10.1038/nmeth.2206">https://doi.org/10.1038/nmeth.2206</a></p>
<p>scTRIP/MosaiCatcher original publication: Sanders, A.D., Meiers, S., Ghareghani, M. et al. Single-cell analysis of structural variations and complex rearrangements with tri-channel processing. Nat Biotechnol 38, 343–354 (2020). <a href="https://doi.org/10.1038/s41587-019-0366-x">https://doi.org/10.1038/s41587-019-0366-x</a></p>
<p>ArbiGent publication: Porubsky, David, Wolfram Höps, Hufsah Ashraf, PingHsun Hsieh, Bernardo Rodriguez-Martin, Feyza Yilmaz, Jana Ebler, et al. 2022. “Recurrent Inversion Polymorphisms in Humans Associate with Genetic Instability and Genomic Disorders.” Cell 185 (11): 1986-2005.e26. <a href="https://doi.org/10.1016/j.cell.2022.04.017">https://doi.org/10.1016/j.cell.2022.04.017</a></p>
<p>scNOVA publication: Jeong, Hyobin, Karen Grimes, Kerstin K. Rauwolf, Peter-Martin Bruch, Tobias Rausch, Patrick Hasenfeld, Eva Benito, et al. 2022. “Functional Analysis of Structural Variants in Single Cells Using Strand-Seq.” Nature Biotechnology, November, 1–13. <a href="https://doi.org/10.1038/s41587-022-01551-4">https://doi.org/10.1038/s41587-022-01551-4</a></p>
</blockquote>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": ".", "features": ["navigation.sections"], "search": "assets/javascripts/workers/search.b8dbb3d2.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"provider": "mike"}}</script>
    
    
      <script src="assets/javascripts/bundle.1e8ae164.min.js"></script>
      
    
  </body>
</html>